nixos/modules/system/boot/resolved.nix

```
@@ -36,6 +38,17 @@ in
       '';
     };
 
+    services.resolved.dns = mkOption {
+      default = null;
```

In this case there's no real difference between empty list and null, lets make it a bit simpler
```suggestion
      default = [ ];
```